### DISCUSSION TITLE
How to use MANAGE_EXTERNAL_STORAGE android permission with MAUI?

### DISCUSSION THREAD

**Discussion by KieranDevvs** [dotnet/maui#6004] [Q&A] [answered]
So I'm wanting to create a simple text editor that can modify files that a user selects on their device. The problem is that `FilePicker` returns a cached file path (it essentially clones the selected file to a directory within the scope of the application and uses that path instead of the original selected file).

While this is an Android API limitation, I have read that for my use case, there is the `MANAGE_EXTERNAL_STORAGE` permission (https://developer.android.com/training/data-storage/manage-all-files).

After I have declared the permission within the manifest and given the application the intent action: 
![image](https://user-images.githubusercontent.com/21192520/163023593-bffc377e-ee40-447b-bb4c-ad3ceeaf7c7d.png)
the `FilePicker` still returns a cached file path.

I can also see that the app does / should have access to the entire device storage.
![image](https://user-images.githubusercontent.com/21192520/163024363-7c2d2685-19d4-4b70-af8b-da21f776d9da.png)

What is the intended usage within MAUI to achieve my goal?

**Comment by KieranDevvs** [comment, marked as answer]
After playing around, i've managed to get raw access to my specified file by going through the Android API's explicitly.
It's starting to look like the `Essentials.FilePicker` doesn't handle the case when you have special storage management permissions.

![image](https://user-images.githubusercontent.com/21192520/163027462-e28509eb-f7a3-423a-8f0b-e7b591c34c24.png)

![image](https://user-images.githubusercontent.com/21192520/163028030-ef816224-4a94-480c-8815-c7cadd272d7e.png)

If anyone has anymore info on this, it would be great to hear from you.

**Reply by jamesmontemagno** [reply]
I mean pretty much Google is going to block all apps that use Read/Write/Manage -> https://developer.android.com/training/data-storage/manage-all-files#all-files-access-google-play so be careful. The file picker uses the file providers in Android to return a trusted shared path so you can copy the contents.

**Reply by KieranDevvs** [reply]
> I mean pretty much Google is going to block all apps that use Read/Write/Manage -> https://developer.android.com/training/data-storage/manage-all-files#all-files-access-google-play so be careful. The file picker uses the file providers in Android to return a trusted shared path so you can copy the contents.

It explicitly says there are use cases that this is intended for, which I do believe a document editor comes under `File Managers` or `Document Management Apps`?
![image](https://user-images.githubusercontent.com/21192520/163142541-fb9dddbf-02b1-4ae9-8d20-de097241eef8.png)

I just find it odd that Android allows this functionality, yet the MAUI `FilePicker` doesn't support this use case. Also, even if my application didn't meet the Google Play store privacy requirements, you assume that all apps will be added to the Play Store, which is really not the case. If the Android API allows it, then MAUI should try to replicate that as best it can? Regardless of the terms and conditions of your app hosting provider (if any)?

I've opened an issue to get this behavior added to `FilePicker` so that applications with specific permissions can do as they intend... After all, that is why the permissions exist in the first place. (https://github.com/dotnet/maui/issues/6015)
